aten/src/ATen/native/Pow.cpp

```diff
@@ -52,12 +53,26 @@ Tensor& pow_(Tensor& base, Scalar alpha) {
 }
 
 Tensor pow(const Tensor& base, const Tensor& exp) {
-  Tensor result = at::empty({0}, base.options());
+  // If the exponent is complex, the result needs to be complex
+  // we can't rely on result_type because it will break current
+  // handling
+  ScalarType dtype = (exp.is_complex() ? exp.scalar_type() : base.scalar_type());
```

Comment:
PR https://github.com/pytorch/pytorch/pull/37098 fixes the type promotion issue more generally so we can get rid of this line here once it is merged :D